consensus/src/dag/rb_handler.rs

```diff
@@ -67,22 +72,13 @@ impl NodeBroadcastHandler {
         self.storage.delete_votes(to_delete)
     }
 
-    fn validate(&self, node: &Node) -> anyhow::Result<()> {
+    fn validate(&self, node: Node) -> anyhow::Result<Node> {
         let current_round = node.metadata().round();
 
-        // round 0 is a special case and does not require any parents
-        if current_round == 0 {
-            return Ok(());
-        }
-
-        let prev_round = current_round - 1;
+        // ensure round is greater than 0. Don't accept a zero-round.
+        ensure!(current_round > 0, NodeBroadcastHandleError::InvalidRound(current_round));
 
         let dag_reader = self.dag.read();
-        // check if the parent round is missing in the DAG
-        ensure!(
```

Comment:
we still need to filter out the lower round right? otherwise we'd fetch all those nodes from gc'ed rounds

Reply:
That is a good point. I can send you a node from a very old round and force you to fetch if it is below the GC level. We should ignore nodes from rounds below the GC level.

Reply:
It's becoming hard to write unit tests without a verifier trait. I will follow-up to introduce that trait and then we can have more unit tests for this case. Added TODO for now.